### PR TITLE
Docs features landing page

### PR DESCRIPTION
**Pull request by godai78**
Type: Docs: Features landing page revamp as recommended in #5703 

This is a kick-off draft for possible further development.

### CODE REVIEW

**Comment by FilipTokarski** [comment]
It looks really nice, can we add a section about Revision History?

**Comment by godai78** [comment]
Revision History added.

File: docs/features/index.md [hunk 1]
```diff
@@ -7,27 +7,113 @@ feedback-widget: false
...
+
+The {@link features/word-count words and characters counter} will help you track progress and control the volume of the content.
+
+Use {@link features/autoformat Markdown syntax} to format content on the go to speed the editing process, employ {@link features/text-transformation automatic text transformations} (also known as autocorrect) and the {@link features/spelling-and-grammar-checking spell checker} to ensure everything is correct. Create multi-language documents and correct them on the go with automatic language detection and {@link features/language text part language} feature.
+
+{@img assets/img/features-spellcheck-multilanguage.png 800 CKEditor 5 spell-checking feature.}
+
+Track your progress and changes done in the content with the {@link features/revision-history revision history} feature. This modern and robust document versioning tools lets you create named versions, compare changes and restore previous document versions at ease, tracking all progress &mdash; also when multiple editors work together.
```
Comment (FilipTokarski) [inline docs/features/index.md:64]: Shouldn't it be `tool` instead of `tools`?

File: docs/features/index.md [hunk 2]
```diff
@@ -7,27 +7,113 @@ feedback-widget: false
...
+
+The [users API](https://ckeditor.com/docs/ckeditor5/latest/features/collaboration/users.html) is used by functions such as [track changes](https://ckeditor.com/docs/ckeditor5/latest/features/collaboration/track-changes/track-changes.html), that allow the users to follow any changes made to the edited document in real time. Accepting or rejecting those changes is done with a single click from a convenient side panel.
+
+Where tracking changes is not enough, the [comments](https://ckeditor.com/docs/ckeditor5/latest/features/collaboration/comments/comments.html) come in, offering perfect collaboration communication platform for writing and editing as a team.
+
+Additionally, CKEditor 5 offers the {@link features/restricted-editing restricted editing mode} where only selected part of the content may be edited by selected users, based on permissions system. And when there is a need for even more control, there is the {@link features/read-only read-only editor mode} that lets the uses access the content, but not edit it.
```
Comment (FilipTokarski) [inline docs/features/index.md:76]: lets the `uses` => lets the `users`

File: docs/features/index.md [hunk 3]
```diff
@@ -7,27 +7,113 @@ feedback-widget: false
...
+
+{@img assets/img/features-collaboration.png 800 CKEditor 5 collaboration features.}
+
+### Customizable user experience
+
+Work as you like it - choose user interface approach from {@link examples/index several preconfigured build}, add or remove features easily to your build with the [online builder](https://ckeditor.com/ckeditor-5/online-builder/) or utilize the {@link builds/guides/development/dll-builds DLL builds}. Then taylor the user interface to your needs with our customizable {@link features/toolbar editor toolbar} arranging features buttons, dropdowns and other items whatever way you need them.
```
Comment (FilipTokarski) [inline docs/features/index.md:82]: several preconfigured `build` => several preconfigured `builds`

File: docs/features/index.md [hunk 4]
```diff
@@ -7,27 +7,113 @@ feedback-widget: false
...
+
+{@img assets/img/features-paste.png 800 CKEditor 5 paste features.}
+
+### Output and exports
+
+Writing and editing content is one thing, sharing it is another. The CKEditor 5 WYSIWYG editor by default produces HTML output, that can be saved into a database. The default output can be also switched to {@link features/markdown Github-flavored Markdown} formatted text.
+
+If you need to share the document, use [export to PDF feature](https://ckeditor.com/docs/ckeditor5/latest/features/export-pdf.html) to produce industry standard, portable, cross-platform final files. If you need to work further on the document, choose the [export to Word](https://ckeditor.com/docs/ckeditor5/latest/features/export-word.html) instead - and keep your comment and changes in the resulting document, ready to be edited further.
```
Comment (FilipTokarski) [inline docs/features/index.md:96]: keep your `comment` and changes => keep your `comments` and changes ?

**Review by FilipTokarski — CHANGES_REQUESTED** [review]
Just some small typos. Other than that, it looks cool 👍

**Review by FilipTokarski — APPROVED** [review]
LGTM.

**Pull request merged**